### ISSUE TITLE
Prometheus incompatibility: labels and label values endpoints filtering works in a different way

### ISSUE THREAD

**Issue by morigs** [VictoriaMetrics/VictoriaMetrics#1398]
Prometheus endpoints `/api/v1/labels` and `/api/v1/label/:name/values` supports optional parameters: `match[]`, `start` and `end`.
Prometheus has the following defaults:
- if `match[]` is empty then include all series
- if `start` and `end` is empty then do not filter by time

But VictoriaMetrics applies the following logic:
- If `match[]` is empty:
  - If `start` and `end` is empty then do not filter by time
  - Else filter by start and end
- If `match[]` is not empty:
  - If `start` and `end` is not empty then filter by them
  - But if `start` and `end` are missing then use current time for `start` and `now-defaultStep` for `end`

Thus VM is acting differently when `math[]` is specified but `start` and `and` are not.
Also for some reason when specifying `start` and `and` explicitly for a long time range latency is much larger than when you just filtering all series ignoring time.

**Example**

*Prometheus*:

```
/api/v1/labels?match[]={__name__!=""}

{
  "status": "success",
  "data": [
    "__name__",
    "alertmanager",
    "alertname",
    "alertstate",
   ...
  ]
}
```

*VictoriaMetrics*:

```
/select/0/prometheus/api/v1/labels?match[]={__name__!=''}

{
  "status": "success",
  "data": []
}
```

**Version** v1.40.0-cluster

**Comment by valyala** [comment]
VictoriaMetrics maintains two indexes for fast lookups of time series by label filters:

* Global index
* Per-day index

These indexes are also used when querying for labels and/or label values via `/api/v1/labels` and `/api/v1/label/<labelName>/values`. When `start` and `end` args are missing, then VictoriaMetrics queries the global index. Otherwise the per-day index is used. When the `start ... end` time range is bigger than one day, then VictoriaMetrics queries all the per-day indexes per each day covered by `start ... end` time range. This can take more time than querying the global index if the `start ... end` time range covers big number of days.

Possible solution: to switch to the global index for big `start ... end` time ranges.

**Comment by morigs** [comment]
Yes, that's correct, global index should be used. But the real reason if the problem is `start` and `end` defaults. They are deliberately set to last five minutes for many endpoints and this behavior is documented.
So I think these defaults should be changed at least in labels endpoints. The downside is this will introduce kind of incompatibility between VictoriaMetrics versions
